packages/bsky/src/views/index.ts

```diff
@@ -453,6 +455,33 @@ export class Views {
     }
   }
 
+  status(did: string, state: HydrationState): StatusView | undefined {
+    const actor = state.actors?.get(did)
+    if (!actor?.status) return
+
+    const { status } = actor
+    const { record, sortedAt } = status
+
+    const maxDuration = 3 * HOUR
+    const expiresAtMs = record.durationMinutes
+      ? sortedAt.getTime() +
+        Math.min(record.durationMinutes * MINUTE, maxDuration)
+      : undefined
+    const expiresAt = expiresAtMs
+      ? new Date(expiresAtMs).toISOString()
+      : undefined
+
+    const isActive = expiresAtMs ? expiresAtMs > Date.now() : undefined
+
+    return {
+      record: record,
+      status: record.status,
+      embed: record.embed,
```

Comment:
Going to want to transform this into the external embed view, so that it contains info useful to the client such as CDN urls.